packages/number-field/src/number-field.css

```diff
@@ -53,3 +52,52 @@ governing permissions and limitations under the License.
         0
     );
 }
+
+:host([invalid]:not([hide-stepper])) #textfield .icon {
+    --mod-textfield-icon-spacing-inline-end-invalid: calc(
+        var(--system-infield-button-top-width) +
+            var(--spectrum-textfield-icon-spacing-inline-end-invalid)
+    );
+}
+
+:host([valid]:not([hide-stepper])) #textfield .icon {
+    --mod-textfield-icon-spacing-inline-end-valid: calc(
+        var(--system-infield-button-top-width) +
+            var(--spectrum-textfield-icon-spacing-inline-end-valid)
+    );
+}
+
+:host(:not([hide-stepper])) #textfield {
+    --mod-stepper-width: calc(
+        var(--spectrum-stepper-height) *
+            var(
+                --mod-stepper-min-width-multiplier,
+                var(--spectrum-text-field-minimum-width-multiplier)
+            ) + var(--spectrum-stepper-button-width) * 2 +
+            var(
+                --mod-stepper-border-width,
+                var(--spectrum-stepper-border-width)
+            ) * 2
+    );
+}
+
+:host([invalid]) .input {
+    padding-inline-end: calc(
```

Comment:
This addresses the icon overlapping the input value/truncation when invalid. This is not an issue for valid.